libraries/rush-lib/src/logic/pnpm/PnpmOptionsConfiguration.ts

```diff
@@ -0,0 +1,282 @@
+   * lifecycle events for the specified NPM dependencies.  This is useful for scripts with poor practices
+   * such as downloading large binaries without retries or attempting to invoke OS tools such as
+   * a C++ compiler.  (PNPM's terminology refers to these lifecycle events as "building" a package;
+   * it has nothing to do with build system operations such as `rush build` or `rushx build`.)
+   * The settings are copied into the `pnpm.neverBuiltDependencies` field of the `common/temp/package.json`
+   * file that is generated by Rush during installation.
+   *
+   * PNPM documentation: https://pnpm.io/package_json#pnpmneverbuiltdependencies
+   */
+  public readonly globalNeverBuiltDependencies: string[] | undefined;
+
+  /**
+   * The `globalAllowedDeprecatedVersions` setting suppresses installation warnings for package
+   * versions that the NPM registry reports as being deprecated.  This is useful if the
+   * deprecated package is an indirect dependency of an external package that has not released a fix.
+   * The settings are copied into the `pnpm.allowedDeprecatedVersions` field of the `common/temp/package.json`
+   * file that is generated by Rush during installation.
+   *
+   * PNPM documentation: https://pnpm.io/package_json#pnpmalloweddeprecatedversions
+   *
+   * If you are working to eliminate a deprecated version, it's better to specify `allowedDeprecatedVersions`
+   * in the package.json file for individual Rush projects.
+   */
+  public readonly globalAllowedDeprecatedVersions: Record<string, string> | undefined;
+
+  /**
+   * (USE AT YOUR OWN RISK)  This is a free-form property bag that will be copied into
+   * the `common/temp/package.json` file that is generated by Rush during installation.
+   * This provides a way to experiment with new PNPM features.  These settings will override
+   * any other Rush configuration associated with a given JSON field except for `.pnpmfile.cjs`.
+   *
+   * USAGE OF THIS SETTING IS NOT SUPPORTED BY THE RUSH MAINTAINERS AND MAY CAUSE RUSH
+   * TO MALFUNCTION.  If you encounter a missing PNPM setting that you believe should
+   * be supported, please create a GitHub issue or PR.  Note that Rush does not aim to
+   * support every possible PNPM setting, but rather to promote a battle-tested installation
+   * strategy that is known to provide a good experience for large teams with lots of projects.
+   */
+  public readonly unsupportedPackageJsonSettings: unknown | undefined;
+
+  private constructor(json: IPnpmOptionsJson, commonTempFolder: string) {
+    super(json);
+    this.pnpmStore = json.pnpmStore || 'local';
+    if (EnvironmentConfiguration.pnpmStorePathOverride) {
+      this.pnpmStorePath = EnvironmentConfiguration.pnpmStorePathOverride;
+    } else if (this.pnpmStore === 'global') {
+      this.pnpmStorePath = '';
+    } else {
+      this.pnpmStorePath = path.resolve(path.join(commonTempFolder, 'pnpm-store'));
```

Comment:
```suggestion
      this.pnpmStorePath = `${commonTempFolder}/pnpm-store`;
```

Is `commonTempFolder` not absolute? Is that way we needed the `path.resolve`?